test/query_test.rb

```diff
@@ -51,14 +51,52 @@ def test_quote
     assert_query quoted, Fauna::Query.quote(quoted)
   end
 
+  def test_lambda
+    assert_equal Fauna::Query.lambda { |a| Fauna::Query.add(a, a) }, {
+      lambda: 'a',
+      expr: { add: [{ var: 'a' }, { var: 'a' }] }
+    }
+
+    lambda = Fauna::Query.lambda do |a|
+      Fauna::Query.lambda do |b|
+        Fauna::Query.lambda { |c| [a, b, c] }
+      end
+    end
+    assert_equal lambda, {
+      lambda: 'a',
+      expr: {
+        lambda: 'b',
+        expr: {
+          lambda: 'c',
+          expr: [{ var: 'a' }, { var: 'b' }, { var: 'c' }]
+        }
+      }
+    }
+
+    # Error in function should not affect future queries.
+    begin
```

Comment:
Yes. Was looking at the code mainly so it wasn't clear. Was expecting an assert that lambda raises, didn't know you were testing for side-effects.